### PR TITLE
Add Keras trackable logic for TF backend (fixes list and dictionary deps)

### PR DESCRIPTION
**Pull request by nkovela1**
This PR adds interfacing logic between Keras tracking and TF trackable logic.

Associated issue: https://github.com/keras-team/keras/issues/18398

- I have created a special subclass KerasAutoTrackable with __setattr__ logic that takes into account Keras's own tracking suite.
- All TF layers now inherit from KerasAutoTrackable, which inherits from the original TF AutoTrackable.
- I have also added associated tests for lists and dictionaries of Trackables in `saved_model_test.py`.

### CODE REVIEW

**Comment by nkovela1** [comment]
There appear to be some issues with modifying tracked list attributes that contain trackables. More TF tracking logic may need to be added here.

**Comment by nkovela1** [comment]
Figured out a solution @fchollet! Ready for review.

File: keras/backend/tensorflow/trackable.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import tensorflow as tf
+
+from keras.utils import tracking
+
+
+class KerasAutoTrackable(tf.__internal__.tracking.AutoTrackable):
```
Comment (fchollet) [inline keras/backend/tensorflow/trackable.py:6]: If the tracking is done strictly by hand, is it better to subclass from `Trackable` to minimize interference from TF?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Pull request merged**